routing/chainview/interface_test.go

```diff
@@ -773,16 +773,107 @@ var interfaceImpls = []struct {
 	name          string
 	chainViewInit chainViewInitFunc
 }{
+	{
+		name: "bitcoind_rpc_polling",
+		chainViewInit: func(rpcInfo rpcclient.ConnConfig,
+			p2pAddr string) (func(), FilteredChainView, error) {
+
+			// Start a bitcoind instance.
+			tempBitcoindDir, err := ioutil.TempDir("", "bitcoind")
+			if err != nil {
+				return nil, nil, err
+			}
+			cleanUp1 := func() {
+				os.RemoveAll(tempBitcoindDir)
+			}
+			rpcPort := rand.Int()%(65536-1024) + 1024
+			bitcoind := exec.Command(
+				"bitcoind",
+				"-datadir="+tempBitcoindDir,
+				"-regtest",
+				"-connect="+p2pAddr,
+				"-txindex",
+				"-rpcauth=weks:469e9bb14ab2360f8e226efed5ca6f"+
+					"d$507c670e800a95284294edb5773b05544b"+
+					"220110063096c221be9933c82d38e1",
+				fmt.Sprintf("-rpcport=%d", rpcPort),
+				"-disablewallet",
+			)
+			err = bitcoind.Start()
+			if err != nil {
+				cleanUp1()
+				return nil, nil, err
+			}
+			cleanUp2 := func() {
+				_ = bitcoind.Process.Kill()
+				_ = bitcoind.Wait()
+				cleanUp1()
+			}
+
+			// Wait for the bitcoind instance to start up.
+			time.Sleep(time.Second)
```

Comment:
Can we tighten this check up here with something like a `wait.Predicate`? Otherwise I fear this'll end up flaking a lot on CI.

Reply:
i havent yet been able to figure out how to completely get rid of the `time.Sleep` even when adding the `wait.NoError`. Seems like needs to happen anywhere before calling `Start` on the chain view. This seems to be the case for the ZMQ connection too so is not specific to this PR so wondering if i can add a todo for now?

Reply:
What about instead doing a mini poll here until we get a valid response from the bitcoind connection?

Reply:
yeah i did try this before but it consistently flakes on my machine. 
I pushed up the patch to demonstrate the flake now but I see it does not happen on github. So it might be a MacOS thing or something specific to my machine or my bitcoind installation...

@Roasbeef , could you try run this test on your machine to see if it flakes?